datafusion/core/src/physical_plan/planner.rs

```diff
@@ -1013,7 +1013,10 @@ impl DefaultPhysicalPlanner {
                         LogicalPlan::TableScan(..) => {
                             self.create_initial_plan(input, session_state).await
                         }
-                        _ => Err(DataFusionError::Plan("SubqueryAlias should only wrap TableScan".to_string()))
+                        LogicalPlan::Filter(..) => {
+                            self.create_initial_plan(input, session_state).await
+                        }
+                        _ => Err(DataFusionError::Plan("SubqueryAlias should only wrap TableScan or Filter".to_string()))
```

Comment:
Ideally we should be able to support wrapping any plan in a subquery with alias (this could also simplify some planning code).
But I got some errors trying this before.
https://github.com/apache/arrow-datafusion/issues/3927

Reply:
I may get your point. You hope to unify alias in the same place
Current, alias can exist two part.
- SubqueryAlias
- Projection Alias
This implementation is different from `Spark`

`Spark Subquery` don't include alias.
```scala
case class Subquery(child: LogicalPlan, correlated: Boolean) extends OrderPreservingUnaryNode {
  override def output: Seq[Attribute] = child.output
  override protected def withNewChildInternal(newChild: LogicalPlan): Subquery =
    copy(child = newChild)
}
```

I think your thought is reasonable. We shouldn't put alias in multiple place.

Reply:
This PR is just a tmp solution for #4014.

When I try to fix `Integrations test`, I add these code to fix it.